### ISSUE TITLE
The line number for a field not found in struct is not indicated correctly.

### ISSUE THREAD

**Issue by houshengbo** [go-yaml/yaml#280]
```
d.terrors = append(d.terrors, fmt.Sprintf("line %d: field %s not found in struct %s", n.line+1, name.String(), out.Type()))
```

This is the line of code, which prints a wrong number of the YAML file, if an unexpected field is found. 
It would be a straightforward fix by replacing n.line+1 with ni.line+1.

**Comment by houshengbo** [comment]
@niemeyer @rogpeppe I believe this is a valid issue, and I have already submitted a PR #281 to resolve it. Please review it.

**Comment by rogpeppe** [comment]
Fixed by #307.
